src/Datadog.Trace.ClrProfiler.Native/module_metadata.h

```diff
@@ -89,23 +143,41 @@ class ModuleMetadata
 
     void SetWrapperMemberRef(const WSTRING& keyIn, const mdMemberRef valueIn)
     {
-        wrapper_refs[keyIn] = valueIn;
+        std::lock_guard<std::mutex> guard(wrapper_mutex);
+        if (wrapper_refs == nullptr)
+        {
+            wrapper_refs = new std::unordered_map<WSTRING, mdMemberRef>();
```

Comment:
same here, why `new` ?

Reply:
In this case yes, we can use `std::unique_ptr` the behavior is the same for both integration methods (CallTarget and CallSite), will do the change ... Although Is not that bad to use `new` here isn't?  XD for modules with any rewriting I will allocate 0 bytes, not true if I'm using smart pointers :D

Reply:
done in 79804dd

Reply:
XD no it's not that bad. I mean, it's just that if it's shared at some point, or we do not handle the lifecycle, it can become a nightmare. I know I can be extreme on that (sorry)
Yeah, this is my thinking:
1. can we use smart pointers first
2. If it's better (for perf reason ...) to use raw pointer, so be it

Reply:
no worries :) I've tested the change with `unique_ptr` and we are ok.